node/bin/src/prover_api/fri_job_manager.rs

```diff
@@ -273,15 +187,70 @@ impl FriJobManager {
             }
         }
 
-        // Deserialize and verify using metadata from the batch.
+        self.verify_proof(&batch_metadata, &proof_bytes, batch_number, prover_id)
+            .await?;
+
+        // We want to ensure we can send the result downstream before we remove the job from queue
+        let permit = self.try_reserve_permit_downstream()?;
+
+        // Remove the job from the assigned map.
+        let Some(removed_job) = self
+            .jobs
+            .complete_job(batch_number, ProverType::Real, prover_id)
+            .await
+        else {
+            // If already removed due to a race
+            // (another submit won), we still return success to keep the API idempotent.
+            tracing::warn!(
```

Comment:
Nit: This will cause a sentry "alert". What exactly would we do in such scenario?